core/extension/gdextension_interface.h

```diff
@@ -2864,6 +2864,11 @@ typedef void (*GDExtensionsInterfaceEditorHelpLoadXmlFromUtf8Chars)(const char *
  */
 typedef void (*GDExtensionsInterfaceEditorHelpLoadXmlFromUtf8CharsAndLen)(const char *p_data, GDExtensionInt p_size);
 
+typedef GDExtensionObjectPtr (*GDExtensionInterfaceCreateGodotInstance)(int p_argc, char *p_argv[], GDExtensionInitializationFunction p_init_func);
+typedef void (*GDExtensionInterfaceDestroyGodotInstance)(GDExtensionObjectPtr p_godot_instance);
+
+GDExtensionObjectPtr gdextension_create_godot_instance(int p_argc, char *p_argv[], GDExtensionInitializationFunction p_init_func);
```

Comment:
Omitting this function declaration works, if we compile Godot as a shared library, because we can look up the symbol by name with dlsym().

However, we also support building LibGodot as a static library, (we use it now on iOS for example, but there is nothing that would prevent using it on other platforms, if desired by the developer). In such case they need the function declaration.

We contemplated adding a separate libgodot.h header for this, but we felt it would be an overkill to add a new header file for a single function declaration.

We could move this declaration into a #ifdef LIBGODOT_STATIC block to make sure that we don't add the declaration if it is not strictly necessary. What do you think?

Reply:
The function declaration would definitely need to be in a separate header - wrapping the declaration in a macro isn't sufficient.

We don't treat `gdextension_interface.h` as "real C code" because not all bindings will be including it with a C compiler - instead they may generate code from it using a custom utility, so we have a number of restrictions on what can be in that file.

(As an aside: this relates to a [recent proposal](https://github.com/godotengine/godot-proposals/discussions/9524) to instead generate `gdextension_interface.h` from an XML or JSON file. This hasn't been discussed by the GDExtension team yet and no one is working on it as far as I know, but it would be a nice way to formalize "`gdextension_interface.h` isn't real C code" and prevent contributors from trying to add stuff there that isn't allowed :-))

Reply:
These have been resolved by the latest update that was built on top of GDExtensionLoader.
The gdextension_interface.h is not touched at all, and we introduced a new libgodot.h that introduces the create and destroy instance function declarations.